src/Platform/Microsoft.Testing.Platform/UI/NonAnsiTerminal.cs

```diff
@@ -0,0 +1,240 @@
+            Append($":{lineNumber}");
+        }
+    }
+
+    public void SetColor(TerminalColor color)
+    {
+        if (_isBatching)
+        {
+            _console.Write(_stringBuilder.ToString());
+            _stringBuilder.Clear();
+        }
+
+        _console.SetForegroundColor(ToConsoleColor(color));
+    }
+
+    public void ResetColor()
+    {
+        if (_isBatching)
+        {
+            _console.Write(_stringBuilder.ToString());
+            _stringBuilder.Clear();
+        }
+
+        _console.SetForegroundColor(_defaultForegroundColor);
+    }
+
+    public void ShowCursor()
+    {
+        // nop
+    }
+
+    public void HideCursor()
+    {
+        // nop
+    }
+
+    public void StartUpdate()
+    {
+        if (_isBatching)
+        {
+            throw new InvalidOperationException("Console is already in batching mode.");
+        }
+
+        _stringBuilder.Clear();
+        _isBatching = true;
+    }
+
+    public void StopUpdate()
+    {
+        _console.Write(_stringBuilder.ToString());
+        _isBatching = false;
+    }
+
+    private ConsoleColor ToConsoleColor(TerminalColor color) => color switch
+    {
+        TerminalColor.Black => ConsoleColor.Black,
+        TerminalColor.DarkRed => ConsoleColor.DarkRed,
+        TerminalColor.DarkGreen => ConsoleColor.DarkGreen,
+        TerminalColor.DarkYellow => ConsoleColor.DarkYellow,
+        TerminalColor.DarkBlue => ConsoleColor.DarkBlue,
+        TerminalColor.DarkMagenta => ConsoleColor.DarkMagenta,
+        TerminalColor.DarkCyan => ConsoleColor.DarkCyan,
+        TerminalColor.DarkWhite => ConsoleColor.White,
+        TerminalColor.Default => _defaultForegroundColor,
+        TerminalColor.Gray => ConsoleColor.Gray,
+        TerminalColor.Red => ConsoleColor.Red,
+        TerminalColor.Green => ConsoleColor.Green,
+        TerminalColor.Yellow => ConsoleColor.Yellow,
+        TerminalColor.Blue => ConsoleColor.Blue,
+        TerminalColor.Magenta => ConsoleColor.Magenta,
+        TerminalColor.Cyan => ConsoleColor.Cyan,
+        TerminalColor.White => ConsoleColor.White,
+        _ => _defaultForegroundColor,
+    };
+
+    public void EraseProgress()
+    {
+        // nop
+    }
+
+    public void RenderProgress(TestWorker?[] progress)
+    {
+        StartUpdate();
+        try
+        {
+            foreach (TestWorker? p in progress)
+            {
+                if (p == null)
+                {
+                    continue;
+                }
+
+                string durationString = $" ({p.Stopwatch.Elapsed.TotalSeconds:F1}s)";
+
+                int passed = p.Passed;
+                int failed = p.Failed;
+                int skipped = p.Skipped;
+
+                string? detail = !RoslynString.IsNullOrWhiteSpace(p.Detail) ? $"- {p.Detail}" : null;
+                Append('[');
+                SetColor(TerminalColor.DarkGreen);
+                Append("✔️");
+                Append(passed.ToString(CultureInfo.CurrentCulture));
```

Comment:
This is why I originally used a simple `+-?` rather than emojis. We don't have any power over formatting the emojis, nor they are consistent across systems, or easily searchable in text (e.g. you have a long list of tests and you want to find failing, it is easier to write `/ -` than `/ ❌` especially when the emoji does not survive some glyph conversion in azdo raw output. 😄

Reply:
I am not sure what JS is doing but most of their frameworks have a nice CLI output:

https://designmodo.com/wp-content/uploads/2016/06/01-javascript-console-code.jpg
https://th.bing.com/th/id/R.8663b3da0350d2e851969f90aa850bf4?rik=Z19u9Vv%2brzeIcg&riu=http%3a%2f%2fdesignmodo.com%2fwp-content%2fuploads%2f2016%2f06%2f01-javascript-console-code.jpg&ehk=z7XaC0qbkaxKTg%2bqcp64ueGR25lfM4k8oMvwjF%2bCyCc%3d&risl=&pid=ImgRaw&r=0

Reply:
They are using x and a different glyph: 

![image](https://github.com/user-attachments/assets/dc780040-5f1c-4cba-a38c-1e6a06b2c722)